pkg/cmd/run/root.go

```diff
@@ -0,0 +1,167 @@
+// Copyright Nitric Pty Ltd.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package run
+
+import (
+	"fmt"
+	"os"
+	"os/signal"
+	"path/filepath"
+	"syscall"
+	"time"
+
+	"github.com/spf13/cobra"
+
+	"github.com/nitrictech/newcli/pkg/build"
+	"github.com/nitrictech/newcli/pkg/provider/run"
+	"github.com/nitrictech/nitric/pkg/membrane"
+	boltdb_service "github.com/nitrictech/nitric/pkg/plugins/document/boltdb"
+	minio "github.com/nitrictech/nitric/pkg/plugins/storage/minio"
+	"github.com/nitrictech/nitric/pkg/worker"
+)
+
+var runCmd = &cobra.Command{
+	Use:   "run [entrypointsGlob]",
+	Short: "run a nitric stack",
+	Long: `Run a nitric stack locally for
+	development/testing
+`,
+	Run: func(cmd *cobra.Command, args []string) {
+		term := make(chan os.Signal, 1)
+		signal.Notify(term, os.Interrupt, syscall.SIGTERM)
+		signal.Notify(term, os.Interrupt, syscall.SIGINT)
+
+		ctx, _ := filepath.Abs(".")
+
+		files, err := filepath.Glob(filepath.Join(ctx, args[0]))
+
+		if err != nil {
+			cobra.CheckErr(err)
+		}
+
+		// Prepare development images
+		if err := build.CreateBaseDev(ctx, map[string]string{
+			"ts": "nitric-ts-dev",
+		}); err != nil {
+			cobra.CheckErr(err)
+		}
+
+		mio, err := run.NewMinio("./.nitric/run", "test-run")
+
+		if err != nil {
+			cobra.CheckErr(err)
+		}
+
+		// start minio
+		if err := mio.Start(); err != nil {
+			cobra.CheckErr(err)
+		}
+
+		if err := mio.Stop(); err != nil {
+			cobra.CheckErr(err)
+		}
+		// Connect dev storage
+		os.Setenv(minio.MINIO_ENDPOINT_ENV, fmt.Sprintf("localhost:%d", mio.GetApiPort()))
+		os.Setenv(minio.MINIO_ACCESS_KEY_ENV, "minioadmin")
+		os.Setenv(minio.MINIO_SECRET_KEY_ENV, "minioadmin")
+		sp, err := minio.New()
+		if err != nil {
+			cobra.CheckErr(err)
+		}
+
+		// Connect dev documents
+		dp, err := boltdb_service.New()
+		if err != nil {
+			cobra.CheckErr(err)
+		}
+
+		// Create a new Worker Pool
+		// TODO: We may want to override GetWorker on the default ProcessPool
+		// For now we'll use the default and expand from there
+		pool := worker.NewProcessPool(&worker.ProcessPoolOptions{
+			MinWorkers: 0,
+			MaxWorkers: 100,
+		})
+
+		// Start a new gateway plugin
+		gw, err := run.NewGateway()
+
+		if err != nil {
+			cobra.CheckErr(err)
+		}
```

Comment:
```suggestion
		cobra.CheckErr(err)
```
no point wrapping this in an err != nil check as this is partly what CheckErr() does

Reply:
fixed.